smithy-codegen-core/src/test/java/software/amazon/smithy/codegen/core/ArtifactMetadataTest.java

```diff
@@ -0,0 +1,42 @@
+package software.amazon.smithy.codegen.core;
+
+import org.junit.jupiter.api.Test;
+import software.amazon.smithy.model.node.ObjectNode;
+
+class ArtifactMetadataTest {
+
+    @Test
+    void toNode() {
+        ArtifactMetadata am = new ArtifactMetadata();
+        am.setId("a");
+        am.setVersion("b");
+        am.setType("c");
+        am.setTimestamp("d");
+
+        ObjectNode node = am.toNode();
+
+        assert node.getStringMember(ArtifactMetadata.ID_TEXT).get().getValue().equals("a");
+        assert node.getStringMember(ArtifactMetadata.VERSION_TEXT).get().getValue().equals("b");
+        assert node.getStringMember(ArtifactMetadata.TYPE_TEXT).get().getValue().equals("c");
+        assert node.getStringMember(ArtifactMetadata.TIMESTAMP_TEXT).get().getValue().equals("d");
+    }
+
+    @Test
+    void fromNode() {
```

Comment:
It would be good to have tests that make sure your required field validation is working. You can write tests and [assert they throw exceptions like this](https://github.com/awslabs/smithy/blob/42faf2f01322ad4adb1cae915b4daec08e0b19b5/smithy-model/src/test/java/software/amazon/smithy/model/node/NodeTest.java#L34).